pkg/kubectl/cmd/apply_test.go

```diff
@@ -113,18 +125,17 @@ func readServiceFromFile(t *testing.T, filename string) *api.Service {
 	return &svc
 }
 
-func annotateRuntimeObject(t *testing.T, originalObj, currentObj runtime.Object, kind string) (string, []byte) {
+func annotateRuntimeObject(t *testing.T, originalObj, currentObj runtime.Object,
+	codec runtime.Codec) (string, []byte) {
 	originalAccessor, err := meta.Accessor(originalObj)
 	if err != nil {
 		t.Fatal(err)
 	}
 
 	originalLabels := originalAccessor.GetLabels()
-	originalLabels["DELETE_ME"] = "DELETE_ME"
-	originalAccessor.SetLabels(originalLabels)
-	original, err := runtime.Encode(testapi.Default.Codec(), originalObj)
-	if err != nil {
-		t.Fatal(err)
+	if originalLabels != nil {
```

Comment:
ok, I dug in and see what is happening. 

first, `annotateRuntimeObject` expects an object that has labels set in its metadata (which is why you were getting a nil panic), and is intended to be paired with `validatePatchApplication`. it adds the "DELETE_ME" label to make sure that reapplying the original object (which is supposed to have some labels set) removes the added label. That's a useful test to run, but it is not the same test as your idempotent apply test. I would not modify `annotateRuntimeObject` to fit your idempotent testcase.

For your idempotent testcase, I would suggest two stub files, one representing clientside data and one serverside data (complete with last-applied annotation). You can just feed the client-side stub to the apply command, and the server-side stub to the server, and verify an empty patch is sent.

With that approach, no changes are needed to the existing apply tests, which is nice. Opened https://github.com/ahakanbaba/kubernetes/pull/1 with those suggestions if you are amenable.